SyncIntegrationTests/xcodebuild.py

```diff
@@ -31,10 +31,12 @@ def test(self, identifier):
             out = subprocess.check_output(
                 args,
                 cwd=os.path.join(here, os.pardir),
-                stderr=subprocess.STDOUT)
+                stderr=subprocess.STDOUT,
+                universal_newlines=True)
         except subprocess.CalledProcessError as e:
             out = e.output
             raise
         finally:
             with open(self.log, 'w') as f:
-                f.writelines(str(out))
+                f.write(out)
```

Comment:
The value of `out` here is a single string, passing it to `writelines` will iterate over it and write each character as a separate  item. Plain `write` should suffice.